lib/logmsg/logmsg.c

```diff
@@ -1863,6 +1864,14 @@ gssize log_msg_get_size(LogMessage *self)
     nv_table_get_memory_consumption(self->payload); // msg.payload (nvtable)
 }
 
+gboolean log_msg_is_source_suspended(LogMessage *self)
+{
+  if (!self->ack_record)
+    return FALSE;
+  LogSource *src = self->ack_record->tracker->source;
+  return !log_source_free_to_send(src);
+}
```

Comment:
I dont like this. The threaded destination driver should cover this case by doing the flush when the queue is empty. Why doesnt that resolve this situation?

Reply:
Ah the reason i don't like this is that LogMessage should not care less about is source being suspended. It should just be a vehicle for transferring messages.

Reply:
@bazsi: I don’t like it either. this patch just wanted to highlight the fact: iw-size and flush-lines/bytes are not independent.

My other suggestion could resolve the issue(calling flush when queue becomes empty).